chainer/distributions/kl_divergence.py

```diff
@@ -0,0 +1,97 @@
+_KLDIVERGENCE = {}
+
+
+def register_kl(dist1, dist2):
+    """Decorator to register KL divergence function.
+
+    This decorator registers a function which computes Kullback-Leibler
+    divergence. This function will be called by `kl_divergence` based on the
+    argument types.
+
+     Args:
+         Dist1(`type`): type of a class inherit from `~chainer.Distribution` to
+         calculate KL divergence.
+         Dist2(`type`): type of a class inherit from `~chainer.Distribution` to
+         calculate KL divergence.
+
+    The decorated functoion takes a instance of `Dist1` and `Dist2` and
+    returns KL divergence value. When `kl_divergence` will be called taking a
+    instance of `Dist1` and `Dist2` after registration, the decorated function
+    will be called.
+
+    .. admonition:: Example
+
+       >>> from chainer import distributions
+       ... @register_kl(Dist1, Dist2)
+       ... def _kl_dist1_dist2(dist1, dist2):
+       ...     return KL
```

Comment:
Make it correct code. You may skip some part with `...`. Also, `register_kl` should be `distributions.register_kl`. I feel it's better to use a code block here instead of the REPL-style document.